### DISCUSSION TITLE
xfreerdp compiled from source does not open a new window?

### DISCUSSION THREAD

**Discussion by Saddingt0n** [FreeRDP/FreeRDP#8460] [Q&A] [answered]
Hi everyone, I tried to compile the project on a machine with the following specs:
- OS: Ubuntu 20.04
- CPU: Intel i7-7700
- RAM: 16 GB

Here's my build process:
1. cd to project directory
2. `cmake -B<build dir> -S<source dir> -DCHANNEL_URBDRC=OFF -WITH_WAYLAND=ON -WITH_WINPR_TOOLS=ON -WITH_X11=ON -WITH_X264=OFF -DCMAKE_BUILD_TYPE=Release`
3. cd to build dir from step 2
4. `make`. After it's done, I got a "xfreerdp" executable in <build dir>/client/X11.

When I run xfreerdp with: 
`./xfreerdp /v:<IP of the remote machine also running Ubuntu 20.04> /u:<username>`

It prompts me to enter password. I enter it then nothing happens. After a few seconds, I got this message and the program exits.

> [14:45:37:791] [41913:41914] [ERROR][com.freerdp.core.connection] - Timeout waiting for activation
> [14:45:37:791] [41913:41914] [ERROR][com.freerdp.core] - rdp_client_connect:freerdp_set_last_error_ex ERRCONNECT_ACTIVATION_TIMEOUT [0x0002001C]

This is not the case with xfreerdp that I installed with: sudo apt install freerdp2-x11
When I run it with the same option as before: `xfreerdp /v:<IP of the remote machine also running Ubuntu 20.04> /u:<username>`

A new window pops up and asks me for the password.

Here are the output of my build and the apt version (I set both log level to debug):
[xfreerdp_my-build.log](https://github.com/FreeRDP/FreeRDP/files/10109195/xfreerdp_my-build.log)
[xfreerdp_apt.log](https://github.com/FreeRDP/FreeRDP/files/10109196/xfreerdp_apt.log)

Is there something wrong with my build process?

**Comment by akallabeth** [comment, marked as answer]
First, for compilation see https://github.com/FreeRDP/FreeRDP/wiki/Compilation
it is best to just do `cmake -GNinja <your config options> -DCMAKE_INSTALL_PREFIX=<your install location> -B<your build dir> -S<your source dir> && cmake --build <your build dir> --target install` to have a working compile installed wherever you like (I do this usually in `/tmp/freerdp` so no files end up in system)

that said, which version are you building? from the log it seems you don´t build from a proper release tar nor from a git checkout

as for you error, what is your remote end?
this happens sometimes with windows servers if they are busy and for `xrdp` we had some reports of a bug in https://github.com/FreeRDP/FreeRDP/issues/8458 which we are still investigating.

**Reply by Saddingt0n** [reply]
> First, for compilation see https://github.com/FreeRDP/FreeRDP/wiki/Compilation it is best to just do `cmake -GNinja <your config options> -DCMAKE_INSTALL_PREFIX=<your install location> -B<your build dir> -S<your source dir> && cmake --build <your build dir> --target install` to have a working compile installed wherever you like (I do this usually in `/tmp/freerdp` so no files end up in system)
> 
> that said, which version are you building? from the log it seems you don´t build from a proper release tar nor from a git checkout
> 
> as for you error, what is your remote end? this happens sometimes with windows servers if they are busy and for `xrdp` we had some reports of a bug in #8458 which we are still investigating.

Thank you! I managed to build it and it runs fine now. However, I want to build FreeRDP on one machine and run the "xfreerdp" binary I got on other machines. But I tried running "xfreerdp" built on machine A (which runs fine) on machine B, it told me that:
`./xfreerdp: symbol lookup error: ./xfreerdp: undefined symbol: client_cli_present_gateway_message`

Both machines are running Ubuntu 20.04. Why do I have this error message? Is there something special I need to do to run xfreerdp built on a different machine?

**Comment by akallabeth** [comment]
@Saddingt0n you need to copy the whole installation directory, not just the binary.
FreeRDP consists of multiple binaries and libraries, if you just copy them around it will not work.
Also for distributing a binary it is recommended to package it properly, see the other compilation methods mentioned.
